cmd/worker/main.go

```diff
@@ -78,6 +85,60 @@ func HandleKeyGeneration(ctx context.Context, t *asynq.Task) error {
 		"keyEDDSA": keyEDDSA,
 	}).Info("Key generation completed")
 
+	// backup vault
+	vault := vaultType.Vault{
+		Name:           p.Name,
+		PublicKeyEcdsa: keyECDSA,
+		PublicKeyEddsa: keyEDDSA,
+		Signers:        partiesJoined,
+		CreatedAt:      timestamppb.New(time.Now()),
+		HexChainCode:   p.ChainCode,
+		KeyShares: []*vaultType.Vault_KeyShare{
+			{
+				PublicKey: keyECDSA,
+				Keyshare:  ecdsaKeyShare,
+			},
+			{
+				PublicKey: keyECDSA,
+				Keyshare:  eddsaKeyShare,
+			},
+		},
+		LocalPartyId:  p.LocalKey,
+		ResharePrefix: "",
+	}
+
+	encryptionPassword := "" // retrieve from vaultCacheItem
+
+	isEncrypted := encryptionPassword != ""
+	vaultStr := vault.String()
```

Comment:
we should marshal vault into protobuf bytes

Reply:
vault.String() marshals the vault internally.